tools/run_pylint.sh

```diff
@@ -18,7 +18,8 @@ SUCCESS=0
 # Ensure we are at the project root directory
 cd $(readlink -f $(dirname $0))/..
 
-for file in `find . -type f -name '*.py'`
+# For .py files which are not in tests or notebook checkpoints
+for file in `find . ! \( -path "./tests/*" -o -path "*/.ipynb_checkpoints/*" \) -type f -name '*.py'`
```

Comment:
I'm worried that find may behave differently on OS X and on Linux here.
How about replacing this with:
```sh
for file in `git ls-files --full-name --deduplicate --exclude-standard '*.py'`
```

This only considers the files that are already part of the repository and not already ignored by our `.gitignore` file.
The drawback is that it can't ignore `tests/` directory as it's part of the repo.
What's the rationale behind ignoring it?

Reply:
Rationale of not pylinting tests was to allow more gradual linter coverage (not doing everything at once) and the fact that tests seemed like a good candidate of what to do in a later iteration. That being said I like git ls-files better.